### PR TITLE
chore: bump go version, update nix

### PR DESCRIPTION
**Pull request by csatib02**
## Overview

- Updated the `Go` version to `1.22`.
- Also updated the dependencies.
- Fixes a [Backlog item](https://github.com/orgs/bank-vaults/projects/2/views/1?filterQuery=nix&pane=issue&itemId=48311457)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,12 @@
 module github.com/bank-vaults/vault-sdk
 
-go 1.19
+go 1.22.2
```
Comment (ramizpolic) [inline go.mod:3]: i think pinning this should unblock the pipeline
```suggestion
go 1.22
```

Comment (csatib02) [reply go.mod]: Didn't work. :/

**Comment by ramizpolic** [comment]
We should disable non-go 1.22 versions. Lets make sure that we document in release notes that SDK only works with go 1.22

**Review by akijakya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**